### ISSUE TITLE
The project doesn't install on my machine

### ISSUE THREAD

**Issue by HusseinBaraja** [muhamed-didovic/vsdown#13]
I have installed both node 18 and yt-dlp but I keep getting this error:
"npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated har-validator@5.1.5: this library is no longer supported
npm warn deprecated har-validator@5.0.3: this library is no longer supported
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm warn deprecated request@2.87.0: request has been deprecated, see https://github.com/request/request/issues/3142
npm warn deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm warn cleanup Failed to remove some directories [
npm warn cleanup   [
npm warn cleanup     'C:\\Users\\Hussain\\AppData\\Roaming\\nvm\\v18.20.5\\node_modules\\vsdown',
npm warn cleanup     [Error: EPERM: operation not permitted, rmdir 'C:\Users\Hussain\AppData\Roaming\nvm\v18.20.5\node_modules\vsdown\node_modules\axios\lib'] {
npm warn cleanup       errno: -4048,
npm warn cleanup       code: 'EPERM',
npm warn cleanup       syscall: 'rmdir',
npm warn cleanup       path: 'C:\\Users\\Hussain\\AppData\\Roaming\\nvm\\v18.20.5\\node_modules\\vsdown\\node_modules\\axios\\lib'
npm warn cleanup     }
npm warn cleanup   ]
npm warn cleanup ]
npm error code 1
npm error path C:\Users\Hussain\AppData\Roaming\nvm\v18.20.5\node_modules\vsdown\node_modules\ytdl-run
npm error command failed
npm error command C:\WINDOWS\system32\cmd.exe /d /s /c node ./install.js
npm error installing youtube-dl...
npm error C:\Users\Hussain\AppData\Roaming\nvm\v18.20.5\node_modules\vsdown\node_modules\download\node_modules\got\index.js:182
npm error                               ee.emit('error', new got.RequestError(err, opts));
npm error                                                ^
npm error
npm error GotError [RequestError]: connect ETIMEDOUT 2001:1a50:11:0:5f:8f:acaa:177:443
npm error     at ClientRequest.<anonymous> (C:\Users\Hussain\AppData\Roaming\nvm\v18.20.5\node_modules\vsdown\node_modules\download\node_modules\got\index.js:182:22)
npm error     at Object.onceWrapper (node:events:632:26)
npm error     at ClientRequest.emit (node:events:517:28)
npm error     at TLSSocket.socketErrorListener (node:_http_client:501:9)
npm error     at TLSSocket.emit (node:events:517:28)
npm error     at emitErrorNT (node:internal/streams/destroy:151:8)
npm error     at emitErrorCloseNT (node:internal/streams/destroy:116:3)
npm error     at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
npm error   code: 'ETIMEDOUT',
npm error   host: 'yt-dl.org',
npm error   hostname: 'yt-dl.org',
npm error   method: 'GET',
npm error   path: '/downloads/latest/youtube-dl.exe',
npm error   protocol: 'https:',
npm error   url: 'https://yt-dl.org/downloads/latest/youtube-dl.exe'
npm error }
npm error
npm error Node.js v18.20.5
npm error A complete log of this run can be found in: C:\Users\Hussain\AppData\Local\npm-cache\_logs\2024-12-02T06_46_35_027Z-debug-0.log
"

Here is a list of what I tried
- power shell with and without admin permission
- CMD with and without admin permission
- Node 18 and node 23
- I suspect the issue arises because the app is trying to download youtube-dl from (https://yt-dl.org/downloads/latest/youtube-dl.exe) which I don't have access to in my location, I tried using a VPN to access the site from a few different location but the same happened in all of them.

![{64D3647C-6BAB-4ABD-8EB4-D3F67448817B}](https://github.com/user-attachments/assets/0f8dab68-e6a9-4373-8bea-746db55f0dc4)

Here is the log file generated by this error:
[2024-12-02T06_46_35_027Z-debug-0.log](https://github.com/user-attachments/files/17970152/2024-12-02T06_46_35_027Z-debug-0.log)

**Comment by muhamed-didovic** [comment]
hey @HusseinBaraja yt-dlp is installed on your machine?

**Comment by HusseinBaraja** [comment]
@muhamed-didovic yes, it's downloaded, installed and added to the environment variables path, I thought I mentioned it in my issue, my bad
I downloaded the latest version as of yesterday

**Comment by muhamed-didovic** [comment]
did you try again?

**Comment by HusseinBaraja** [comment]
Hi, @muhamed-didovic sorry for the late reply
unfortunately it still doesn't work
Here is a screen shot of my latest try

![{1B02240C-79D3-49EC-B916-B9EB06A00A2A}](https://github.com/user-attachments/assets/14077a40-7670-45d0-800c-e2848ceb8068)

**Comment by HusseinBaraja** [comment]
Hello @muhamed-didovic, any updates?

**Comment by muhamed-didovic** [comment]
hey @HusseinBaraja I see from image that it still wants to install youtube-dl...

**Comment by HusseinBaraja** [comment]
So, how can I solve it?

@muhamed-didovic

**Comment by muhamed-didovic** [comment]
hit me on the discord: memo23

**Comment by iamwatchdogs** [comment]
I have faced a similar issue, I couldn't figure out even though I switched back to node:18 using `nvm`. So, I cleaned up everything I had done and tried various things by creating a couple of Docker images and playing around with containers and coming back tweaking the Dockerfile to create newer images and repeat the cycle. Finally, I did reach a Dockerfile that can create the image that can work out just fine.

> [!TIP]
> Even though you don't know much about Docker, just follow along and you do just fine... Just make sure you have Docker on your system, if you don't just download [Docker Desktop](https://www.docker.com/products/docker-desktop/).

- Start by cloning the repo with depth of 1 i.e., with the last recent commit.
 
  ```bash
  git clone --depth 1 https://github.com/muhamed-didovic/vsdown.git
  ```
- Create a `Dockerfile` with following config:
 
  ```dockerfile
  FROM ubuntu:latest
  
  ENV DEBAIN_FRONTEND=noninteractive
  
  RUN apt-get update -y && \
      apt-get upgrade -y
  
  RUN apt-get install -y \
      software-properties-common \
      wget \
      curl \
      gnupg2 \
      lsb-release \
      ca-certificates \
      && apt-get clean \
      && rm -rf /var/lib/apt/lists/*
  
  RUN apt-get install -y python3
  
  RUN curl -fsSL https://deb.nodesource.com/setup_18.x | bash - \
      && apt-get install -y nodejs
  
  RUN wget https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp -O /usr/local/bin/youtube-dl && \
      chmod +x /usr/local/bin/youtube-dl
  
  WORKDIR /app
  
  COPY package*.json .
  RUN npm ci
  
  COPY . .
  
  RUN chmod +x lib/cli.js
  
  RUN npm link
  
  ENTRYPOINT ["vsdown"]
  ```
- Now build an image out of it.
 
  ```bash
  docker build -t vsdown .
  ```

> [!IMPORTANT]
> If you're doing this for the first time, then it may take more time. But eventually, it will become faster as it caches a few layers. So, if you want to tweak it a bit, go ahead...

- Now use `docker run` to create an instant container for running the application.

  ```bash
  docker run -it --rm -v $(pwd):/downloads -w /downloads vsdown [args]
  ```
- If you feel like that's redundant to us it in such a way, you can just add the following to your `~/.bashrc`,
 
  ```bash
  vsdown() {
    docker run -it --rm -v $(pwd):/downloads -w /downloads vsdown "$@"
  }
  ```
- Then you can just use `vsdown` right on your terminal just like any other command. These changes, i.e., `vsdown` will become accessible from a new session of the terminal, but if you wish to use it the right way with your current terminal session, then you gotta...
 
  ```bash
  source ~/.bashrc
  ```

Hope this is helpful !!!...
